crates/core/component/shielded-pool/src/spend/action.rs

```diff
@@ -117,10 +120,27 @@ impl TryFrom<pb::SpendBody> for Body {
             .try_into()
             .context("malformed rk")?;
 
+        // `EncryptedBackref` must have 0 or `ENCRYPTED_BACKREF_LEN` bytes.
+        let encrypted_backref: EncryptedBackref;
+        if proto.encrypted_backref.len() == ENCRYPTED_BACKREF_LEN {
+            let bytes: [u8; ENCRYPTED_BACKREF_LEN] = proto
+                .encrypted_backref
+                .try_into()
+                .map_err(|_| anyhow::anyhow!("invalid encrypted backref"))?;
+            encrypted_backref = EncryptedBackref::try_from(bytes)
+                .map_err(|_| anyhow::anyhow!("invalid encrypted backref"))?;
+        } else if proto.encrypted_backref.len() == 0 {
+            encrypted_backref = EncryptedBackref::try_from([0u8; ENCRYPTED_BACKREF_LEN])
```

Comment:
This caught my eye because we get a proto with a missing/empty encrypted backref and validate it into a `EncryptedBackref` domain type that will return `false` on `EncryptedBackref::is_empty`.

Reply:
good catch